samtranslator/utils/traverse.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Callable, Dict, List
+
+
+class Traverse:
```

Comment:
Let's decouple traversal algorithm from action performed while traversing, in that way it would be easier to add more actions if we need them in the future